src/Compilers/CSharp/Portable/CodeGen/EmitStatement.cs

```diff
@@ -1050,10 +1053,27 @@ private void EmitCatchBlock(BoundCatchBlock catchBlock)
             }
 
             EmitBlock(catchBlock.Body);
-
             _builder.CloseLocalScope();
         }
 
+        private static bool IsFilterConstantFalse(BoundCatchBlock catchBlock)
+        {
+            // Depending on compilation options, BoundLiteral can be wrapped by BoundSequence or BoundSequencePointExpression
+            if(catchBlock.ExceptionFilterOpt != null)
```

Comment:
Consider flipping this condition around: `if (... == null) { return false; }` (with correct formatting)